### PR TITLE
Rle read and parse

### PR DESCRIPTION
**Pull request by YaffaLevy**

### CODE REVIEW

File: src/main/java/levy/gameoflife/RleParser.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package levy.gameoflife;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class RleParser {
+    int width;
+    int height;
+    private int[][] grid;
```
Comment (andrewoid) [inline src/main/java/levy/gameoflife/RleParser.java]: Why are these member variables?

File: src/main/java/levy/gameoflife/RleParser.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        }
+        return grid;
+    }
+
+    private int fillGrid(int runCount, char tag, int col, int row) {
+        for (int i = 0; i < runCount; i++) {
+            if (tag == 'o') {
+                grid[row][col++] = 1;
+            } else if (tag == 'b') {
+                grid[row][col++] = 0;
```
Comment (andrewoid) [inline src/main/java/levy/gameoflife/RleParser.java]: It's already 0

File: src/test/java/levy/gameoflife/GameOfLifeTest.java [hunk 3]
```diff
@@ -1,11 +1,36 @@
 package levy.gameoflife;
 
-import levy.gameoflife.GameOfLife;
 import org.junit.jupiter.api.Test;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 class GameOfLifeTest {
 
+    @Test
+    public void rleParser() {
```
Comment (andrewoid) [inline src/test/java/levy/gameoflife/GameOfLifeTest.java]: loadFromRle()

File: src/main/java/levy/gameoflife/RleParser.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+public class RleParser {
+    public int getWidth() {
+        return width;
+    }
+
+    public int getHeight() {
+        return height;
+    }
+
```
Comment (andrewoid) [inline src/main/java/levy/gameoflife/RleParser.java]: Methods should be below member variables

File: src/main/java/levy/gameoflife/RleParser.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    private int width;
+    private int height;
+    private int[][] grid;
+
+    public int[][] parse(String rleContent) {
+        String[] lines = rleContent.split("\n");
+        Pattern headerPattern = Pattern.compile("x\\s*=\\s*(\\d+),\\s*y\\s*=\\s*(\\d+)");
```
Comment (andrewoid) [inline src/main/java/levy/gameoflife/RleParser.java:13]: What does this do?

File: src/main/java/levy/gameoflife/RleParser.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            line = line.trim();
+            if (line.startsWith("#")) {
+                continue;
+            }
+            Matcher matcher = headerPattern.matcher(line);
+            if (matcher.find() && !gridInitialized) {
+                this.width = Integer.parseInt(matcher.group(1));
+                this.height = Integer.parseInt(matcher.group(2));
+                this.grid = new int[height][width];
+                gridInitialized = true;
```
Comment (andrewoid) [inline src/main/java/levy/gameoflife/RleParser.java]: You don't need this Boolean, you know if grid != Null

File: src/main/java/levy/gameoflife/RleParser.java (hunk 5 above)
Comment (YaffaLevy) [reply src/main/java/levy/gameoflife/RleParser.java:13]: this looks for this format( x = <number>, y = <number>) to find the dimensions

**Review by andrewoid — APPROVED** [review]

**Pull request merged**